opm/simulators/linalg/DILU.hpp

```diff
@@ -113,7 +134,11 @@ class MultithreadDILU : public PreconditionerWithUpdate<X, Y>
         if (use_multithreading) {
             parallelUpdate();
         } else {
-            serialUpdate();
+            if (split_matrix_) {
+                serialSplitUpdate();
```

Comment:
The method names serial[Split]Update and parallelUpdate may be confusing since they both can be used in an MPI parallel setting, but just depend on the number of threads per process. I suggest taking the opportunity to rename them to for example singleThreaded[Split]Update and multiThreadedUpdate.